.github/actions/lint-tags/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file This action checks if the tags used in posts exist in the
+ * yaml file that defines all supported tags.
+ */
+
+function run() {
+  const core = require('@actions/core');
+  const fs = require('fs');
+  const glob = require('glob');
+  const matter = require('gray-matter');
+  const yaml = require('js-yaml');
+  const path = require('path');
+
+  const tagsYmlPath = path.join(
+    process.cwd(),
+    core.getInput('tags_yml', {required: true}),
+  );
+  /** @type {{[tag: string]: any}} */
+  const tags = yaml.load(fs.readFileSync(tagsYmlPath, 'utf-8')) || {};
+  const postsGlob = path.join(
+    process.cwd(),
+    core.getInput('posts_path', {required: true}),
+    '{**/*,*}.md',
+  );
+  const postsPaths = glob.sync(postsGlob, {});
+  const ignoreTags = core.getMultilineInput('ignore_tags', {required: false});
+  let tagErrorCount = 0;
+
+  if (Object.keys(tags).length === 0) {
+    core.setFailed('No tags from tags yaml found');
+  }
+
+  // Add tags to ignore to our tags master list
+  for (const ignoreTag of ignoreTags) {
+    tags[ignoreTag] = true;
+  }
+
+  // Validate all posts to see if there are any invalid tags
+  for (const postPath of postsPaths) {
+    const fileData = fs.readFileSync(postPath, 'utf8');
+    const data = matter(fileData).data;
+
+    // If `data.tags` is a string rather than an array, turn it into an array
+    if (typeof data.tags === 'string') {
+      data.tags = [data.tags];
+      // If `data.tags` isn't a string or an array set it to an empty array
+    } else if (!Array.isArray(data.tags)) {
+      data.tags = [];
+    }
+
+    for (const tag of data.tags) {
+      if (!(tag in tags)) {
+        tagErrorCount++;
+        core.warning(
+          `ERROR IN FILE: ${postPath.replace(
+            process.cwd(),
+            '',
+          )}, "${tag}" is not a valid tag`,
+        );
+      }
+    }
+  }
+
+  if (tagErrorCount !== 0) {
+    core.setFailed(`Found ${tagErrorCount} instance(s) of invalid tags used`);
+  }
+}
+
+// Install dependencies
+require('child_process').spawnSync('npm', ['install'], {
+  cwd: __dirname,
+});
```

Comment:
Do you need this? I thought GH actions installed whatever was in the package.json automatically?